src/flag_gems/ops/normal.py

```diff
@@ -0,0 +1,80 @@
+import logging
+
+import torch
+import triton
+
+from ..utils import pointwise_dynamic
+from ..utils.random_utils import philox_cuda_seed_offset
+from ..utils.shape_utils import broadcast_shapes, volume
+from .randn import randn_kernel
+
+UNROLL = 4
+
+
+@pointwise_dynamic(
+    is_tensor=[True, True, True], promotion_methods=[(0, 1, 2, "DEFAULT")]
+)
+@triton.jit
+def transform_func_tensor_tensor(val, std, mean):
+    return val * std + mean
+
+
+@pointwise_dynamic(
+    is_tensor=[True, True, False], promotion_methods=[(0, 1, 2, "DEFAULT")]
+)
+@triton.jit
+def transform_func_tensor_float(val, std, mean):
+    return val * std + mean
+
+
+@pointwise_dynamic(
+    is_tensor=[True, False, True], promotion_methods=[(0, 1, 2, "DEFAULT")]
+)
+@triton.jit
+def transform_func_float_tensor(val, std, mean):
+    return val * std + mean
+
+
+@pointwise_dynamic(
+    is_tensor=[True, False, False], promotion_methods=[(0, 1, 2, "DEFAULT")]
+)
+@triton.jit
+def transform_func_float_float(val, std, mean):
+    return val * std + mean
+
+
+def normal_distribution(mean, std, *, generator=None):
+    shape = broadcast_shapes([mean.shape, std.shape])
+    out = torch.empty(shape, device=mean.device, dtype=torch.float32)
+    N = volume(shape)
+    grid_fn = lambda meta: (triton.cdiv(N, meta["BLOCK"] * UNROLL),)
+
+    increment = triton.cdiv(N, UNROLL)
+    philox_seed, philox_offset = philox_cuda_seed_offset(increment)
+    with torch.cuda.device(mean.device):
+        randn_kernel[grid_fn](out, N, philox_seed, philox_offset)
+    return out
+
+
+def normal_tensor_tensor(mean, std, *, generator=None):
+    logging.debug("GEMS NORMAL_TENSOR_TENSOR")
+    out = normal_distribution(mean, std)
+    return transform_func_tensor_tensor(out, std, mean)
```

Comment:
normal needs to launch two kernels. suggest fusing them together.

Reply:
OK，will try that later